### PR TITLE
Cancel order #100

### LINKED ISSUE #100: User Story 35, User cancels an order
```
As a registered user
When I visit an order's show page
If the order is still "pending", I see a button or link to cancel the order
When I click the cancel button for an order, the following happens:
- Each row in the "order items" table is given a status of "unfulfilled"
- The order itself is given a status of "cancelled"
- Any item quantities in the order that were previously fulfilled have their quantities returned to their respective merchant's inventory for that item.
- I am returned to my profile page
- I see a flash message telling me the order is now cancelled
- And I see that this order now has an updated status of "cancelled"
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #103

depends on #120

depends on #102

### PR DESCRIPTION
**Pull request by stoic-plus**
closes #100 

Canceling order functionaltity - added class method to OrderItem and Item

* OrderItem has method to find all fulfilled order item - item id's and their quantities
  * Calls Item.return_inventory to add those quantities back to respective items

* Item.return_inventory takes the item_ids and adds the quantity to inventory 

Page Interaction
* Only shows button to cancel if order is pending 
* Changed profile_order_path to only show order info for that one order (was showing info for all orders for a user)

### CODE REVIEW

File: app/controllers/profile/orders_controller.rb [hunk 1]
```diff
@@ -2,8 +2,20 @@ class Profile::OrdersController < ApplicationController
...
+    order = Order.find(params[:id])
+
+    order.status = "cancelled"
+    order.save
+    OrderItem.return_inventory_for(order.id)
+    OrderItem.unfulfill_items_for(order.id)
+
+    flash[:success] = "Order has been cancelled."
+    redirect_to profile_path
+  end
```
Comment (nicovigil1) [inline app/controllers/profile/orders_controller.rb]: clever use of the destroy method!

File: app/models/Item.rb [hunk 2]
```diff
@@ -17,4 +17,10 @@ def self.popular_items(top_or_bottom, amount)
         .select("items.*, sum(quantity) as item_count")
   end
 
+  def self.return_inventory(items_and_quantities)
```
Comment (nicovigil1) [inline app/models/Item.rb]: maybe singular? "item & quantity"

File: app/models/order_item.rb [hunk 3]
```diff
@@ -5,6 +5,29 @@ class OrderItem < ApplicationRecord
...
+        quantity: c_i.last,
+        price: item.price,
+        order_id: order_id,
+        item_id: item.id
+      )
+    end
+  end
+
+  def self.unfulfill_items_for(order_id)
+    where(order_id: order_id).update_all(fulfilled: false, updated_at: Time.now)
```
Comment (nicovigil1) [inline app/models/order_item.rb]: update_all? nice.
